### ISSUE TITLE
pyopenms 2.7.0: static method(s) broken

### ISSUE THREAD

**Issue by uweschmitt** [OpenMS/OpenMS#6470]
Tested with 2.7.0 and 2.6.0 on Mac M1 with Python 3.7:

```
import pyopenms
print(pyopenms.__version__)
p = pyopenms.Param()
pyopenms.TransformationModelBSpline.getDefaultParameters(p)
```

results in 
```
2.7.0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor 'getDefaultParameters' requires a 'pyopenms.pyopenms_7.TransformationModelBSpline' object but received a 'pyopenms.pyopenms_4.Param'
```
vs
```
2.6.0
```

So in `pyopenms` version `2.7.0` the method `getDefaultParameters` was not wrapped as a static method anymore.

The `.pxd` file looks fine, might be an issue with `autowrap`.

**Comment by uweschmitt** [comment]
The `.pxd` is not fine, did not check thoroughly.

The commit https://github.com/OpenMS/OpenMS/commit/4f46877020a296671450d26c1f5234b17ee29e06#diff-8b161dfe5d97f4522bdd50087bf5fcbb0ba673f945254caef202eaf24cec34ab broke this.

**Comment by uweschmitt** [comment]
The bug is still present in `2.8.0` and the current development branch.

**Comment by jpfeuffer** [comment]
Thanks for reporting! It means it is specific for this class/method?

Or do you think there is something wrong in general with static methods, e.g. in autowrap?

**Comment by uweschmitt** [comment]
The title "[fix] fixed python errors so all tests pass under language level 3" of the commit 4f46877020a296671450d26c1f5234 indicates that this either was an attempt to fix a bug or is just a left-over from this attempt.

So there is nothing broken in `autowrap`, other static methods seem to work.

As far as I could see this is the only class affected, but better double check if there other "left-overs". At least the commit which introduced the bug does not affect other `.pxd` files.

**Comment by poshul** [comment]
> 

I've submitted a PR https://github.com/OpenMS/OpenMS/pull/6477 reverting these changes (IIRC these two static methods had an issue with the specific version of autowrap we were using when we set the python language level to 3). Tests all pass locally

**Comment by poshul** [comment]
Fixed with #6477
